src/mame/sony/news_020_mmu.h

```diff
@@ -0,0 +1,108 @@
+// license:BSD-3-Clause
+// copyright-holders:Brice Onken
+
+#ifndef MAME_SONY_NEWS_020_MMU_H
+#define MAME_SONY_NEWS_020_MMU_H
+
+#pragma once
+
+class news_020_mmu_device : public device_t, public device_memory_interface
+{
+public:
+	news_020_mmu_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	template <typename... T>
+	void set_bus_error_callback(T &&...args) { m_bus_error.set(std::forward<T>(args)...); }
+
+	// MMU entry accessors
+	uint32_t mmu_entry_r(offs_t offset, uint32_t mem_mask);
+	void mmu_entry_w(offs_t offset, uint32_t data, uint32_t mem_mask);
+	void clear_entries(offs_t offset, uint8_t data);
+
+	// MMU control register accessors
+	void set_mmu_enable(bool enabled);
+	void set_rom_enable(bool enabled);
+
+	// Bus accessors for CPU to use when going from CPU->HB
+	uint32_t hyperbus_r(offs_t offset, uint32_t mem_mask, bool is_supervisor);
+	void hyperbus_w(offs_t offset, uint32_t data, uint32_t mem_mask, bool is_supervisor);
+
+protected:
+	virtual void device_start() override;
+	virtual space_config_vector memory_space_config() const override;
+
+private:
+	// Helper class for reading page table entries
+	struct news_020_pte
+	{
+		news_020_pte(uint32_t pte_bits) : pte(pte_bits) {}
```

Comment:
This doesn’t need to be in the header – you can throw it in an anonymous namespace in the .cpp file so it isn’t unnecessarily compiled downstream.